kv/src/main/java/com/test/kv/lib/storage/Memory.java

```diff
@@ -0,0 +1,197 @@
+package com.test.kv.lib.storage;
+
+import java.time.Instant;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.sun.org.apache.bcel.internal.generic.INSTANCEOF;
+import com.sun.tools.jdi.ConcreteMethodImpl;
+import com.test.kv.exceptions.KVKeyDoesNotExistsException;
+import com.test.kv.interfaces.Storage;
+
+/**
+ * Implementation of Storage -- In memory store with LRU
+ */
+public class Memory<KeyType, ValueType> implements Storage<KeyType, ValueType> {
```

Comment:
yeah, looks like so, I'm getting too used to `package.Type` calling pattern in Go :trollface: